File: 392-IsSubsequence/392-IsSubsequence.cpp
// Last updated: 5/5/2025, 3:48:00 PM
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int j = 0;
        int i = 0;
        while (j < s.length()){
            if (i == t.length()){
                return false;
            }
            if (s[j] == t[i]){
                j++;
                i++;
            }
            else{
                i++;
            }
        }
        return true;
    }
};

File: 392-IsSubsequence/392-IsSubsequence.py
# Last updated: 11/8/2025, 12:51:32 PM
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        j = 0
        while i != len(s):
            if j == len(t):
                return False

            if s[i] == t[j]:
                i += 1
                j += 1

            else:
                j += 1

        return True
